packages/paste-core/components/page-header/src/PageHeaderDetails.tsx

```diff
@@ -0,0 +1,38 @@
+import { Box, safelySpreadBoxProps } from "@twilio-paste/box";
+import type { BoxProps } from "@twilio-paste/box";
+import type { HTMLPasteProps } from "@twilio-paste/types";
+import * as React from "react";
+
+export interface PageHeaderDetailsProps extends HTMLPasteProps<"div"> {
+  children?: React.ReactNode;
+  /**
+   * Overrides the default element name to apply unique styles with the Customization Provider
+   * @default 'PAGE_HEADER_DETAILS'
+   * @type {BoxProps['element']}
+   * @memberof PageHeaderDetailsProps
+   */
+  element?: BoxProps["element"];
+}
+
+const PageHeaderDetails = React.forwardRef<HTMLDivElement, PageHeaderDetailsProps>(
+  ({ element = "PAGE_HEADER_DETAILS", children, ...props }, ref) => {
+    return (
+      <Box
+        {...safelySpreadBoxProps(props)}
+        ref={ref}
+        element={element}
+        marginBottom="space130"
+        display="grid"
+        gridTemplateColumns="auto auto 1fr"
+        gridTemplateRows="auto-fit"
+        gridTemplateAreas='"detail detail detail" "prefix heading actions" "meta meta meta" "paragraph paragraph paragraph"'
```

Comment:
todo: These should all have a max width of size70 in the default variant only:

- Prefix + Heading together (or just Heading is probably fine if this makes the API too wonky)
- Meta
- Paragraph

The compact variant doesn't need width constraints.

Reply:
related question: Is there anyway to prioritize Button text not wrapping if it's super long, and force Heading to wrap instead?

<img width="1249" alt="Screenshot 2023-12-28 at 12 12 51 PM" src="https://github.com/twilio-labs/paste/assets/3821840/99e4d19b-3235-49ac-9f44-a8f7342aa3fd">

Reply:
This is an extreme example but shows what happens when heading and button text both overflow. I think it handles it pretty well, what do you think?

<img width="1234" alt="image" src="https://github.com/twilio-labs/paste/assets/75342690/81db9b57-4e10-4ec8-89b9-a8581ee6797d">

Reply:
In situations where you have obnoxiously large screens like me, would a percentage max width be better than a physical size?

<img width="2777" alt="image" src="https://github.com/twilio-labs/paste/assets/368249/3704c32a-a55f-41cf-af27-f05544d6be6a">

Reply:
yaa ok this isn't too bad.

Reply:
Percentage does some weird stuff to the button:
![Screenshot 2024-01-03 at 4 27 00 PM](https://github.com/twilio-labs/paste/assets/3821840/983402a3-eb18-4c9a-a284-54edec9faf6e)

I guess I'm not opposed to bumping up the max width. But also, most pages aren't going to have so much text that wraps (except for paragraph text), so it's going to all look like it's very much to the left anyway.

Reply:
Yeah, I'm not convinced on the percetage. I think once we have a collection of real examples I'll feel less weird about it.